onnxruntime/core/optimizer/noop_elimination.cc

```diff
@@ -40,7 +32,13 @@ bool NoopElimination::SatisfyCondition(const Graph& graph, const Node& node, con
     return false;
   }
 
-  const auto* initializer = graph_utils::GetConstantInitializer(graph, node.InputDefs()[input0_is_initializer ? 0 : 1]->Name());
+  const auto& op_type = node.OpType();
+  if ((op_type == "Sub" || op_type == "Div") && !input1_is_initializer) {
```

Comment:
either input1_is_initializer or input0_is_initializer should be allowed right? .i.e.

if ((op_type == "Sub" || op_type == "Div") && (input1_is_initializer || input0_is_initializer)) ?

Reply:
x+0 == 0+x and x*1 == 1*x, but 0-x != x-0, and 1/x != x/1, so for Sub and Div, the constant 0/1 can be only input1.